File: 0_100/019_Remove_Nth_Node_From_End_Of_List/Solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* dummy = new ListNode(0);
        dummy->next = head;
        ListNode* a = dummy, *b = head;
        while (b) {
            b = b->next;
            if (n-- <= 0)
                a = a->next;
        }
        a->next = a->next->next;
        return dummy->next;
    }
};

File: 0_100/019_Remove_Nth_Node_From_End_Of_List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode b = head, dummy = new ListNode(0);
        dummy.next = head;
        ListNode a = dummy;
        while (b != null) {
            b = b.next;
            if (n-- <= 0)
                a = a.next;
        }
        a.next = a.next.next;
        return dummy.next;
    }
}